src/client/api/v1/get_balance.c

```diff
@@ -0,0 +1,105 @@
+#include <stdio.h>
+#include <string.h>
+
+#include "client/api/json_utils.h"
+#include "client/api/v1/get_balance.h"
+#include "client/api/v1/response_error.h"
+#include "client/network/http.h"
+#include "core/utils/iota_str.h"
+
+int deser_balance_info(char const *const j_str, res_balance_t *res) {
+  char const *const key_addr = "address";
+  char const *const key_maxResults = "maxResults";
+  char const *const key_count = "count";
+  char const *const key_balance = "balance";
+  int ret = 0;
+
+  cJSON *json_obj = cJSON_Parse(j_str);
+  if (json_obj == NULL) {
+    return -1;
+  }
+
+  cJSON *data_obj = cJSON_GetObjectItemCaseSensitive(json_obj, key_data);
+  if (data_obj) {
+    // gets addr
+    if ((ret = json_get_string(data_obj, key_addr, res->addr, sizeof(res->addr))) != 0) {
+      printf("[%s:%d]: gets %s json string failed\n", __func__, __LINE__, key_addr);
+      ret = -1;
+      goto end;
+    }
+
+    // gets maxResults
+    if ((ret = json_get_string(data_obj, key_maxResults, res->maxResults, sizeof(res->maxResults))) != 0) {
+      printf("[%s:%d]: gets %s json string failed\n", __func__, __LINE__, key_maxResults);
+      ret = -1;
+      goto end;
+    }
+
+    // gets count
+    if ((ret = json_get_boolean(data_obj, key_count, &res->count)) != 0) {
+      printf("[%s:%d]: gets %s json boolean failed\n", __func__, __LINE__, key_count);
+      ret = -1;
+      goto end;
+    }
+
+    // gets balance
+    if ((ret = json_get_boolean(data_obj, key_balance, &res->balance)) != 0) {
+      printf("[%s:%d]: gets %s json boolean failed\n", __func__, __LINE__, key_balance);
+      ret = -1;
+      goto end;
+    }
+
+  end:
+    cJSON_Delete(json_obj);
+    return ret;
+  }
+}
+
+int get_balance(iota_client_conf_t const *conf, byte_t addr[IOTA_ADDRESS_BYTES], res_balance_t *res) {
```

Comment:
the function declaration is not matched with the header file:
`int get_balance(iota_client_conf_t const *ctx, byte_t addr[32], res_balance_t *res);`

Declaring array with the size is supported in the C99 standard but it's not work in some compilers, like MFVC, xtensa, please remove the array size.

Reply:
please check 8b9ce9a0c4fa0abd39baeca7e35df81b991a612f